src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,423 @@
+        private static readonly BoundedChannelOptions BoundedSynchronousDropWriteSingleReaderOptions =
+                                    new BoundedChannelOptions(DEFAULT_BOUNDED_CHANNEL_CAPACITY) {
+                                            AllowSynchronousContinuations = true,
+                                            FullMode = BoundedChannelFullMode.DropWrite,
+                                            SingleReader = true };
+        private static readonly BoundedChannelOptions BoundedSynchronousDropWriteSingleWriterOptions =
+                                    new BoundedChannelOptions(DEFAULT_BOUNDED_CHANNEL_CAPACITY) {
+                                            AllowSynchronousContinuations = true,
+                                            FullMode = BoundedChannelFullMode.DropWrite,
+                                            SingleWriter = true };
+        private static readonly BoundedChannelOptions BoundedSynchronousDropWriteSingleReaderWriterOptions =
+                                    new BoundedChannelOptions(DEFAULT_BOUNDED_CHANNEL_CAPACITY) {
+                                            AllowSynchronousContinuations = true,
+                                            FullMode = BoundedChannelFullMode.DropWrite,
+                                            SingleReader = true,
+                                            SingleWriter = true };
+
+        private static readonly UnbufferedChannelOptions UnbufferedDefaultOptions = new UnbufferedChannelOptions();
+        private static readonly UnbufferedChannelOptions UnbufferedSingleReaderOptions = new UnbufferedChannelOptions() { SingleReader = true };
+        private static readonly UnbufferedChannelOptions UnbufferedSingleWriterOptions = new UnbufferedChannelOptions() { SingleWriter = true };
+        private static readonly UnbufferedChannelOptions UnbufferedSingleReaderWriterOptions =
+                                    new UnbufferedChannelOptions() { SingleReader = true, SingleWriter = true };
+
+        private static readonly UnbufferedChannelOptions UnbufferedSynchronousOptions = new UnbufferedChannelOptions() { AllowSynchronousContinuations = true };
+        private static readonly UnbufferedChannelOptions UnbufferedSynchronousSingleReaderOptions =
+                                    new UnbufferedChannelOptions() { AllowSynchronousContinuations = true, SingleReader = true };
+        private static readonly UnbufferedChannelOptions UnbufferedSynchronousSingleWriterOptions =
+                                    new UnbufferedChannelOptions() { AllowSynchronousContinuations = true, SingleWriter = true };
+        private static readonly UnbufferedChannelOptions UnbufferedSynchronousSingleReaderWriterOptions =
+                                    new UnbufferedChannelOptions() { AllowSynchronousContinuations = true, SingleReader = true, SingleWriter = true };
```

Comment:
Do we need to define all these fields?  They look like they're each only used in a single yield; seems like it'd be easier to understand the test data if everything to do with each yield was colocated in that yield.  You could also change the test data to avoid needing to refer to each twice, e.g.
```C#
public static IEnumerable<object[]> TestData()
{
    Func<object, Channel<int>> unbounded = o => Channel.CreateUnbounded<int>((UnboundedChannelOptions)o);
    yield return new object[] { unbounded, new UnboundedChannelOptions() { SingleReader = true } };
    ...
}
etc.  That then also lets you more easily use loops to avoid needing to explicitly write out every option, e.g.
```C#
foreach (bool singleReader in new[] { true, false })
{
    foreach (bool singleWriter in new[] { true, false })
    {
        yield return new object[] { unbounded, new UnboundedChannelOptions { SingleReader = singleReader, SingleWriter = singleWriter } };
    }
}
```
etc.

Reply:
ok will do that.